packages/dev/loaders/src/glTF/2.0/Extensions/interactivityFunctions.ts

```diff
@@ -0,0 +1,211 @@
+            className: convertedType,
+        };
+    } else {
+        return configObject.value;
+    }
+}
+
+function convertConfiguration(gltfBlock: IKHRInteractivity_Node, definition: IKHRInteractivity, id: string): IFlowGraphBlockConfiguration {
+    const converted: IFlowGraphBlockConfiguration = {};
+    const configurationList: IKHRInteractivity_Configuration[] = gltfBlock.configuration ?? [];
+    for (const configObject of configurationList) {
+        if (configObject.id === "customEvent") {
+            const customEvent = definition.customEvents && definition.customEvents[configObject.value];
+            if (!customEvent) {
+                throw new Error(`/extensions/KHR_interactivity/nodes/${id}: Unknown custom event: ${configObject.value}`);
+            }
+            converted.eventId = customEvent.id;
+            converted.eventData = customEvent.values.map((v) => v.id);
+        } else if (configObject.id === "variable") {
+            const variable = definition.variables && definition.variables[configObject.value];
+            if (!variable) {
+                throw new Error(`/extensions/KHR_interactivity/nodes/${id}: Unknown variable: ${configObject.value}`);
+            }
+            converted.variableName = variable.id;
+        } else if (configObject.id === "path") {
+            // Convert from a GLTF path to a reference to the Babylon.js object
+            const pathValue = configObject.value as string;
+            converted.path = {
+                path: pathValue,
+                className: "FGPath",
+            };
+        } else {
+            converted[configObject.id] = convertValueWithType(configObject, definition, `/extensions/KHR_interactivity/nodes/${id}`);
+        }
+    }
+    return converted;
+}
+
+function convertBlock(id: number, gltfBlock: IKHRInteractivity_Node, definition: IKHRInteractivity): ISerializedFlowGraphBlock {
+    const className = gltfToFlowGraphTypeMap[gltfBlock.type];
+    if (!className) {
+        throw new Error(`/extensions/KHR_interactivity/nodes/${id}: Unknown block type: ${gltfBlock.type}`);
+    }
+    const uniqueId = id.toString();
+    const config = convertConfiguration(gltfBlock, definition, uniqueId);
+    const metadata = gltfBlock.metadata;
+    const dataInputs: ISerializedFlowGraphConnection[] = [];
+    const dataOutputs: ISerializedFlowGraphConnection[] = [];
+    const signalInputs: ISerializedFlowGraphConnection[] = [];
+    const signalOutputs: ISerializedFlowGraphConnection[] = [];
+    return {
+        className,
+        config,
+        uniqueId,
+        metadata,
+        dataInputs,
+        dataOutputs,
+        signalInputs,
+        signalOutputs,
+    };
+}
+
+/**
+ * Converts a glTF Interactivity Extension to a serialized flow graph.
+ * @param gltf the interactivity data
+ * @returns a serialized flow graph
+ */
+export function convertGLTFToJson(gltf: IKHRInteractivity): ISerializedFlowGraph {

```

Comment:
If the description is right, the function name should be changed. The function name states that this function converts glTF (which might already be a JSON) to a JSON. but this is specific to FlowGrph. 
I noticed this function is really "internal" as it is not exported in index.ts as well. maybe we can mark it as `@internal`?